crates/stateless/src/execution_witness.rs

```diff
@@ -0,0 +1,78 @@
+/*
+This file has been copied from alloy-rpc-types-debug.
+
+The reason we have it is because alloy-rpc-types-debug is not no_std compatible.
+Checking the Cargo.toml file shows that it enables "std" for alloy-primitives.
+Currently it does not seem to need "std", so one option is to disable it.
+
+One could alternatively argue that the type may not belong in rpc-debug-types and
+may belong somewhere like alloy-primitives and re-exported in rpc-types-debug.
+
+Reason being, with all variants of statelessness (when shipped),
+there is an `ExecutionWitness` that comes with the block, so its
+not just used for debugging.
+
+*/
+use alloc::vec::Vec;
+use alloy_primitives::Bytes;
+use serde::{Deserialize, Serialize};
+
+/// Represents the execution witness of a block. Contains an optional map of state preimages.
+#[derive(Clone, Debug, Default, PartialEq, Eq, Serialize, Deserialize)]
+pub struct ExecutionWitness {
+    /// List of all hashed trie nodes preimages that were required during the execution of
+    /// the block, including during state root recomputation.
+    pub state: Vec<Bytes>,
+    /// List of all contract codes (created / accessed) preimages that were required during
+    /// the execution of the block, including during state root recomputation.
+    pub codes: Vec<Bytes>,
+    /// List of all hashed account and storage keys (addresses and slots) preimages
+    /// (unhashed account addresses and storage slots, respectively) that were required during
+    /// the execution of the block.
+    pub keys: Vec<Bytes>,
```

Comment:
do we still need this?

Reply:
This was copied verbatim from `alloy-rpc-types` -- I think we would need it in the future if we want to do the pre state verification checks upfront.

For this file, we can also make alloy-rpc-types-debug no_std and then remove it

Reply:
opened https://github.com/alloy-rs/alloy/issues/2399